code-challenges.test.js

```diff
@@ -13,57 +13,136 @@
+// // const makeCaps = (arr) => {
+// //   let calc = arr.map(e => e.charAt(0).toUpperCase() + e.slice(1)).join('')
+// //   return calc;
+// // }
+// //Pseudocode: 
+// //declare a function that takes an array 
+// //create a local variable that maps thru the array
+// //with each value access the first index and capitalize the letter 
+// //then concatenate the remainder of the word
+// //my test kept failing. It seems to return randomNums1 to cap the first letter but not randNums2. Why would it work on one array and not the other.?...well I had toUppercase()instead of toUpperCase() then I removed the join method and it worked
+// const makeCaps = (array) => {                         
+//   let eachItem = array.map((value) => {               
+//     return value[0].toUpperCase() + value.substring(1)
+//   })                                                   
+//     return eachItem
+// }
+
+// // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
+
+// // a) Create a test with expect statements for each of the variables provided.
+// //Pseudocode: describe  a function that takes in a string and logs the index of the first vowel.
+// //iterate through a string .map() and 
+// //return the location of the first vowel. 
+// // a describe method that lists the name of the function OR naming of the particular test.
+describe("spellCheck", () => {
+// a test/it method, that checks if the function spellCheck returns the index of the first vowel
+// Ran test and was a good Red LIght at the expect statement FAIL  ./code-challenges.test.js
+
+  it("function that takes in a string and logs the index of the first vowel.", () => {
+    
+    expect(spellCheck(vowelTester1)).toEqual(1);
+    expect(spellCheck(vowelTester2)).toEqual(0);
+    expect(spellCheck(vowelTester3)).toEqual(2);
+  });
+});
 
 const vowelTester1 = "learn"
 // Expected output: 1
 const vowelTester2 = "academy"
 // Expected output: 0
 const vowelTester3 = "challenges"
 // Expected output: 2
-
-
 // b) Create the function that makes the test pass.
+//Pseudocode: function that takes in an string and iterates thru to find the first instance of a vowel and returns the index of that vowel
+//declare a local variable to store the vowels. 
+//for loop iterate thru the string to find the index of the matching string to the array
+//Refactor ideas: turn the vowels into an array and iterate 
+// const spellCheck = (string) => {
+// // return string.indefOf((value) => value == firstVowel)
+// let littles = string.split('');
+// for(i=0; i<littles.length; i++){
+//   if(littles[i] == 'a' || littles[i] == 'e' || littles[i] == 'i' || littles[i] == 'o' ||littles[i] == 'u'){
+//     return i;
+//   }}
+// }
+
+
+
+//// function spellCheck(str){
+//   let test = str.search(/[a,e,i,o,u]/g)
+//   console.log("flag",test)
+//// }
+
+//// spellCheck(vowelTester1)         attempted Regex did not work, yet....
```

Comment:
love that word yet...remember to believe in your ability to work through it. I would suggest looking at what output you are receiving from using the search method then figuring out if you can use the output in your logic